Mathlib/Algebra/Ring/Equiv.lean

```diff
@@ -814,6 +814,33 @@ theorem symm_trans_self (e : R ≃+* S) : e.symm.trans e = RingEquiv.refl S :=
 
 end RingEquiv
 
+namespace RingEquiv
+
+variable [NonAssocSemiring R] [NonAssocSemiring S]
+
+/-- If a ring homomorphism has an inverse, it is a ring isomorphism. -/
+@[simps]
+def ofRingHom (f : R →+* S) (g : S →+* R) (h₁ : f.comp g = RingHom.id S)
```

Comment:
This is essentially [RingHom.inverse](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Ring/Equiv.html#RingHom.inverse), I am not sure we want to duplicate it. Do we do it for other types of homs?

Reply:
> This is essentially [RingHom.inverse](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Ring/Equiv.html#RingHom.inverse), I am not sure we want to duplicate it. Do we do it for other types of homs?

I used [AlgEquiv.ofAlgHom](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Equiv.html#AlgEquiv.ofAlgHom) to construct `mapAlgEquiv`, and then I wanted to construct `mapRingEquiv` in a similar way. Therefore, I wrote `RingEquiv.ofRingHom` by imitating `AlgEquiv.ofAlgHom`.